packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -1598,4 +1697,36 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
     const autoComplete = !this.props.disabled && this.props.autoComplete === 'on';
     return autoComplete ? (this.props.allowFreeform ? 'inline' : 'both') : 'none';
   }
+
+  private _isPendingOption(item: IComboBoxOption): boolean {
+    return item && item.index === this.state.currentPendingValueValidIndex;
+  }
+
+  private _getSelectedKeys(
+    defaultSelectedKey: string | number | string[] | number[] | undefined,
+    selectedKey: string | number | string[] | number[] | undefined
+  ): string[] | number[] {
+
+    let retKeys: string[] | number[] = [];
+
+    if (defaultSelectedKey) {
+      if (defaultSelectedKey instanceof Array) {
+        retKeys = defaultSelectedKey;
+      } else if (typeof defaultSelectedKey === 'string') {
+        retKeys = [defaultSelectedKey as string];
+      } else if (typeof defaultSelectedKey === 'number') {
+        retKeys = [defaultSelectedKey as number];
+      }
+    } else if (selectedKey) {
```

Comment:
Shouldn't this be an if here so that we'll fix up both the defaultSelectedKey and selectedKey? I know at most one should be used, but will we get into any trobule if they both aren't arrays when we go to process them later?

Reply:
To eliminate potential confusion, I added the following comment in the code. This function handles the general logic which takes precedence.  Caller decides whether they pass in defaultSelectedKey or not.  In the constructor, the caller does, but it doesn't when receiving new props.

  /**
   * Given default selected key(s) and selected key(s), return the selected keys(s).
   * When default selected key(s) are available, they take precedence and return them instead of selected key(s).
   * 
   * @returns No matter what specific types the input parameters are, always return an array of
   *  either strings or numbers instead of premitive type.  This normlization makes caller's logic easier.
   */